app/database/operator/handlers/base_handler.ts

```diff
@@ -0,0 +1,457 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {MM_TABLES} from '@constants/database';
+import DatabaseManager from '@database/manager';
+import {
+    isRecordAppEqualToRaw,
+    isRecordCustomEmojiEqualToRaw,
+    isRecordGlobalEqualToRaw,
+    isRecordRoleEqualToRaw,
+    isRecordServerEqualToRaw,
+    isRecordSystemEqualToRaw,
+    isRecordTermsOfServiceEqualToRaw,
+} from '@database/operator/comparators';
+import {
+    prepareAppRecord,
+    prepareCustomEmojiRecord,
+    prepareGlobalRecord,
+    prepareRoleRecord,
+    prepareServersRecord,
+    prepareSystemRecord,
+    prepareTermsOfServiceRecord,
+} from '@database/operator/prepareRecords/general';
+import {Database, Q} from '@nozbe/watermelondb';
+import Model from '@nozbe/watermelondb/Model';
+
+import {
+    BatchOperationsArgs,
+    DatabaseInstance,
+    HandleEntityRecordsArgs,
+    HandleIsolatedEntityArgs,
+    PrepareForDatabaseArgs,
+    PrepareRecordsArgs,
+    ProcessInputsArgs,
+    RawValue,
+    RecordPair,
+} from '@typings/database/database';
+import {IsolatedEntities, OperationType} from '@typings/database/enums';
+
+import DataOperatorException from '../../exceptions/data_operator_exception';
+import DatabaseConnectionException from '../../exceptions/database_connection_exception';
+import {
+    getRangeOfValues,
+    getRawRecordPairs,
+    getUniqueRawsBy,
+    hasSimilarUpdateAt,
+    retrieveRecords,
+} from '../utils/general';
```

Comment:
non blocking but can we make this not relative paths? maybe use an alias like we did with `@database`, perhaps something like `@dbexceptions` and `@dbutils`

Reply:
Updated